src/cargo/util/toml.rs

```diff
@@ -741,6 +745,18 @@ fn unique_names_in_targets(targets: &[TomlTarget]) -> Result<(), String> {
     Ok(())
 }
 
+/// Will check a list of build targets, and make sure the target names are unique within a vector.
+/// If not, the name of the offending build target is returned.
+fn unique_build_targets(targets: &Vec<Target>) -> Result<(), &str> {
+    let mut seen = HashSet::new();
+    for v in targets.iter().map(|e| e.name()) {
```

Comment:
Ah yeah here I think you want to use `src_path` instead, as `.name()` is checked by the function above